.github/workflows/build-and-push.yaml

```diff
@@ -133,8 +135,10 @@ jobs:
         name: Fetch and Set Environment Variables from Secret Manager
         id: deploy-secrets
         run: |
+          # Get appropriate secret name
+          SECRET_NAME=$(gcloud secrets list --filter="labels.environment:${{ steps.deploy-info.outputs.environment_name }} AND labels.deployment:${{ steps.deploy-info.outputs.deployment }}" --project=${{ steps.deploy-info.outputs.project }} --format="value(name)")
           # The secret value must be a single-line JSON string.
-          SECRET_JSON=$(gcloud secrets versions access latest --secret=${{ steps.deploy-info.outputs.appname }}-${{ steps.deploy-info.outputs.environment_name }} --project=${{ steps.deploy-info.outputs.project }})
+          SECRET_JSON=$(gcloud secrets versions access latest --secret=$SECRET_NAME --project=${{ steps.deploy-info.outputs.project }})
```

Comment:
You shouldn't need the `--project` arg because the service account Github secret contains the project ID and is already authenticated at this point